hackathon_site/dashboard/frontend/src/components/general/ProductOverview/ProductOverview.js

```diff
@@ -0,0 +1,228 @@
+import React from "react";
+import Typography from "@material-ui/core/Typography";
+import Link from "@material-ui/core/Link";
+import Button from "@material-ui/core/Button";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import LaunchIcon from "@material-ui/icons/Launch";
+import InputLabel from "@material-ui/core/InputLabel";
+import Chip from "@material-ui/core/Chip";
+import styles from "./ProductOverview.module.scss";
+import Alert from "@material-ui/lab/Alert";
+
+import { Formik } from "formik";
+
+const createQuantityList = (number) => {
+    let entry = [];
+
+    for (let i = 1; i <= number; i++) {
+        entry.push(
+            <MenuItem key={i} role="quantity" value={i.toString()}>
+                {i}
+            </MenuItem>
+        );
+    }
+
+    return entry;
+};
+
+export const CartForm = ({
+    availableQuantity,
+    handleSubmit,
+    handleChange,
+    requestFailure,
+    values: { quantity },
+}) => {
+    return (
+        <>
+            {requestFailure && (
+                <Alert className={styles.alert} variant="filled" severity="error">
+                    {requestFailure.message}
+                </Alert>
+            )}
+            <form className={styles.form} onSubmit={handleSubmit}>
+                <FormControl variant="outlined" className={styles.formControl}>
+                    <InputLabel>Qty</InputLabel>
+                    <Select
+                        value={quantity}
+                        role="selecter"
+                        onChange={handleChange}
+                        label="Qty"
+                        name="quantity"
+                    >
+                        {createQuantityList(availableQuantity)}
+                    </Select>
+                </FormControl>
+                <Button
+                    variant="contained"
+                    className={styles.cartButton}
+                    type="submit"
+                    onClick={handleSubmit}
+                    disableElevation
+                >
+                    ADD TO CART
+                </Button>
+            </form>
+        </>
+    );
+};
+
+export const EnhancedCartForm = ({
+    handleSubmit,
+    requestFailure,
+    availableQuantity,
+}) => {
+    const onSubmit = (formikValues) => {
+        handleSubmit(formikValues.quantity);
+    };
+
+    return (
+        handleSubmit && (
```

Comment:
No need to have `handleSubmit &&` - you can assume that it would be defined

Reply:
From another comment above, since you're trying to capture the case where the form isn't necessary, move this check out into the `ProductOverview` component.